mesonbuild/modules/fpga.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2017 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from . import ModuleReturnValue
+from . import ExtensionModule
+from . import noKwargs
+from . import permittedKwargs
+from .. import build
+import os
+
+class BitstreamTarget(build.CustomTarget):
+    def __init__(self, name, subdir, kwargs):
+        super().__init__(name, subdir, kwargs)
+
+class SimulationTarget(build.CustomTarget):
+    def __init__(self, name, subdir, kwargs):
+        super().__init__(name, subdir, kwargs)
+
+class AnalysisTarget(build.CustomTarget):
+    def __init__(self, name, subdir, kwargs):
+        super().__init__(name, subdir, kwargs)
+
+class FpgaLibraryTarget(build.CustomTarget):
+    def __init__(self, name, subdir, packages, kwargs):
+        super().__init__(name, subdir, kwargs)
+        self.packages = packages
+        self.libs_dependencies=[]
+        #if 'depends' in kwargs:
```

Comment:
__[flake8]__
* [E265] block comment should start with '# '

<!-- Comment by [SideCI](https://sideci.com) -->